demos/http/http_demo_plaintext/http_demo_plaintext.c

```diff
@@ -0,0 +1,361 @@
+
+/**
+ * @brief Send an HTTP request based on a specified method and path.
+ *
+ * @param[in] tcpSocket TCP socket.
+ * @param[in] pMethod The HTTP request method.
+ * @param[in] pPath The Request-URI to the objects of interest.
+ *
+ * @return #HTTP_SUCCESS if successful.
+ */
+static HTTPStatus_t _sendHttpRequest( HTTPNetworkContext_t * pContext,
+                                      const char * pMethod,
+                                      const char * pPath )
+{
+    HTTPStatus_t returnStatus = HTTP_SUCCESS;
+    HTTPRequestHeaders_t requestHeaders = { 0 };
+    HTTPRequestInfo_t requestInfo = { 0 };
+    HTTPTransportInterface_t transport = { 0 };
+    HTTPResponse_t response = { 0 };
+
+    SdkLog( "Sending HTTP %s request to %s%s\r\n",
+            pMethod, SERVER, pPath );
+
+    /* Initialize the request object. */
+    requestInfo.method = pMethod;
+    requestInfo.methodLen = sizeof( pMethod ) - 1;
+    requestInfo.pPath = pPath;
+    requestInfo.pathLen = sizeof( pPath ) - 1;
+
+    requestHeaders.pBuffer = userBuffer;
+    requestHeaders.bufferLen = USER_BUFFER_LENGTH;
+
+    returnStatus = HTTPClient_InitializeRequestHeaders( &requestHeaders,
+                                                        &requestInfo );
+
+    if( returnStatus == HTTP_SUCCESS )
+    {
+        /* Define the transport interface. */
+        transport.recv = transportRecv;
+        transport.send = transportSend;
+        transport.pContext = pContext;
+
+        /* Initialize the response object. */
+        response.pBuffer = userBuffer;
+        response.bufferLen = USER_BUFFER_LENGTH;
+
+        /* Send the request and receive the response. */
+        returnStatus = HTTPClient_Send( &transport,
+                                        &requestHeaders,
+                                        requestBodyBuffer,
+                                        REQUEST_BODY_TEXT_LENGTH,
+                                        &response );
+    }
+
+    if( returnStatus == HTTP_SUCCESS )
+    {
+        /* Print the response. */
+        SdkLog( "Response Headers\n%.*s\n",
+                response.headersLen,
+                response.pHeaders );
+        SdkLog( "Response Body\n%.*s",
+                response.bodyLen,
+                response.pBody );
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Entry point of demo.
+ */
+int main()
+{
+    HTTPStatus_t returnStatus = HTTP_SUCCESS;
+    HTTPNetworkContext_t socketContext = { 0 };
+
+    /* Establish TCP connection. */
+    socketContext.tcpSocket = connectToServer( SERVER, PORT );
+
+    if( socketContext.tcpSocket == -1 )
+    {
+        returnStatus = HTTP_NETWORK_ERROR;
+    }
+
+    /*********************** Send HTTPS request. ************************/
+
+    /* The client is now connected to the server. This example will send a
+     * GET, HEAD, PUT, and POST request. */
+
+    #ifdef GET_PATH
```

Comment:
These pre-processor guards are not required as these macros will always be defined (There are defaults defined at the beginning of the file)